pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -196,6 +197,60 @@ func (s *DiscoveryServer) endpointz(w http.ResponseWriter, req *http.Request) {
 	_, _ = fmt.Fprint(w, "\n{}]\n")
 }
 
+// SyncedVersions shows what resourceVersion of a given resource has been acked by Envoy.
+type SyncedVersions struct {
+	ProxyID         string `json:"proxy,omitempty"`
+	ClusterVersion  string `json:"cluster_acked,omitempty"`
+	ListenerVersion string `json:"listener_acked,omitempty"`
+	RouteVersion    string `json:"route_acked,omitempty"`
+}
+
+func (s *DiscoveryServer) distributedVersions(w http.ResponseWriter, req *http.Request) {
+	if resourceID := req.URL.Query().Get("resource"); resourceID != "" {
+		knownVersions := make(map[string]string)
+		results := []SyncedVersions{}
+		adsClientsMutex.RLock()
+		for _, con := range adsClients {
+			con.mu.RLock()
+			if con.modelNode != nil {
+				// TODO: handle skipped nodes
+				results = append(results, SyncedVersions{
+					ProxyID:         con.modelNode.ID,
+					ClusterVersion:  s.getResourceVersion(con.ClusterNonceAcked, resourceID, knownVersions),
+					ListenerVersion: s.getResourceVersion(con.ListenerNonceAcked, resourceID, knownVersions),
+					RouteVersion:    s.getResourceVersion(con.RouteNonceAcked, resourceID, knownVersions),
+				})
+			}
+			con.mu.RUnlock()
+		}
+		adsClientsMutex.RUnlock()
+
+		out, err := json.MarshalIndent(&results, "", "    ")
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			_, _ = fmt.Fprintf(w, "unable to marshal syncedVersion information: %v", err)
+			return
+		}
+		w.Header().Add("Content-Type", "application/json")
```

Comment:
nit: is there some constants for these wellknown headers defined in net/http?

Reply:
It doesn't appear so: https://golang.org/pkg/net/http/ .  However, content type can be omitted altogether, and auto-detected by the DetectContentType function.  That sounds ill advised, personally...

Reply:
sgtm.